fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -98,8 +102,19 @@ public IndexWriter getWriter(LuceneAnalyzerWrapper analyzerWrapper) throws IOExc
                             .setMergeScheduler(new ConcurrentMergeScheduler() {
                                 @Override
                                 public synchronized void merge(final MergeSource mergeSource, final MergeTrigger trigger) throws IOException {
-                                    LOGGER.trace("mergeSource={}", mergeSource);
-                                    super.merge(mergeSource, trigger);
+                                    state.context.addLuceneIndexMergeHook(state.index.getName(), () -> {
+                                        try {
+                                            LOGGER.trace(KeyValueLogMessage.of("Basic index mergeSource",
+                                                    LuceneLogMessageKeys.MERGE_SOURCE, mergeSource,
+                                                    LuceneLogMessageKeys.MERGE_TRIGGER, trigger,
+                                                    LuceneLogMessageKeys.INDEX_NAME, state.index.getName(),
+                                                    LuceneLogMessageKeys.INDEX_SUBSPACE, state.indexSubspace));
+                                            super.merge(mergeSource, trigger);
```

Comment:
I think there are a few ways in which this is a little delicate vis-à-vis the scheduling of other operations:

1. We close the index writer in a commit hook. So, as written, the merges are executed before the writer is closed, though I have a proposal that would put the merges in a vanilla commit hook, which would mean that that's no longer the case. That means that right now, if closing the writer ever triggers a merge (which, I'm not sure if it does or not), then those merges won't ever get run. In the future, it gets a little more subtle
2. If there's a range delete (i.e., a `deleteRecordsWhere`) over a directory, then I think the merges associated with that delete aren't cleaned up properly
3. Ideally, it would be good to verify that the `LuceneMergeScheduler` is okay with `merge()` being a delayed call. I have no idea what semantics it expects of its implementations